tensor2tensor/models/transformer.py

```diff
@@ -388,7 +394,7 @@ def forced_logits():
         ret = tf.cond(
             tf.less(i, partial_targets_length), forced_logits, lambda: ret)
       return ret, cache
-
+    decode_length_decide_end=features['decode_length_decide_end'] if 'decode_length_decide_end' in features else False
```

Comment:
I don't see the value of having this in features. How about getting it from `decode_hparams` and piping it through?

Reply:
Since decode_hparams is used to build the decoding static graph, the decoding process can't adapt the decode length for different inputs.
More specifically, we build a raw Tensorflow session to do the continuously decoding online, so we can't afford to build the graph again and again. 
Actually my plan is to refactor our raw Tensorflow session decoder and send another pull request later.
Thanks a lot!